components/browser/awesomebar/src/main/java/mozilla/components/browser/awesomebar/layout/DefaultSuggestionViewHolder.kt

```diff
@@ -38,13 +38,27 @@ internal sealed class DefaultSuggestionViewHolder {
         private val descriptionView = view.findViewById<TextView>(R.id.mozac_browser_awesomebar_description).apply {
             setTextColor(awesomeBar.styling.descriptionTextColor)
         }
-        private val iconView = view.findViewById<ImageView>(R.id.mozac_browser_awesomebar_icon)
+        private val iconView: ImageView = view.findViewById(R.id.mozac_browser_awesomebar_icon)
+        private val iconIndicatorView: ImageView = view.findViewById(R.id.mozac_browser_awesomebar_icon_indicator)
 
         override fun bind(suggestion: AwesomeBar.Suggestion, selectionListener: () -> Unit) {
             val title = if (suggestion.title.isNullOrEmpty()) suggestion.description else suggestion.title
 
             iconView.setImageBitmap(suggestion.icon)
 
+            if (suggestion.flags.contains(AwesomeBar.Suggestion.Flag.BOOKMARK)) {
+                iconIndicatorView.apply {
+                    setImageDrawable(suggestion.indicatorIcon)
+                    visibility = View.VISIBLE
+                }
+            }
+            if (suggestion.flags.contains(AwesomeBar.Suggestion.Flag.OPEN_TAB)) {
+                iconIndicatorView.apply {
+                    setImageDrawable(suggestion.indicatorIcon)
+                    visibility = View.VISIBLE
+                }
+            }
```

Comment:
Since `indicatorIcon` is nullable I think we no longer need to check the `flags` and can just assign it whenever there is one? :)

Reply:
Also note that those views get recycled, so in the case we do not have an icon we should set the view explicitly to `gone` again.

Reply:
Done.